apps/server-monolith/app/models/user.rb

```diff
@@ -0,0 +1,145 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: pods
+#
+#  id                                                                         :bigint           not null, primary key
+#  bio("Bio" means Biography in social media.)                                :string(140)
+#  deleted_at                                                                 :datetime
+#  display_name                                                               :string           not null
+#  external_avatar_url                                                        :string
+#  last_block_ids                                                             :json             not null
+#  last_pod_username                                                          :string
+#  suspended_at(the date when the user was suspended)                         :datetime
+#  suspended_reason(enumeration value for the reason for the user suspension) :integer          default(0)
+#  type                                                                       :enum             not null
+#  username                                                                   :string           not null
+#  created_at                                                                 :datetime         not null
+#  updated_at                                                                 :datetime         not null
+#
+# Indexes
+#
+#  index_pods_on_deleted_at           (deleted_at)
+#  index_pods_on_lower_username_text  (lower((username)::text)) UNIQUE
+#  index_pods_on_type                 (type)
+#
+class User < Pod
+  delegate :timezone, to: :config
+  delegate :locale, to: :config
+
+  has_one :authentication, class_name: 'Users::Authentication', dependent: :destroy, inverse_of: :user
+  has_one :personal_pod, class_name: 'Pod', foreign_key: 'id', dependent: :destroy, inverse_of: false
+  has_one :owner, class_name: 'User', foreign_key: 'id', dependent: :destroy, inverse_of: false
+
+  has_many :group_members, -> { enabled }, class_name: 'Groups::Member', dependent: :destroy, inverse_of: :user
+  has_many :groups, through: :group_members
+  has_many :owned_group_members, -> { where(role: :owner).enabled }, class_name: 'Groups::Member', dependent: :destroy, inverse_of: :user
+  has_many :owned_groups,
+    # don't destroy the user if it is owner of a group
+    through: :owned_group_members, source: :group, dependent: :restrict_with_exception
+
+  ## FederatedIdentity
+  has_many :federated_identities, class_name: 'Accounts::FederatedIdentity',
+    foreign_key: :accounts_user_id, inverse_of: :user, dependent: :destroy
+
+  attr_accessor :email, :password, :confirmed_at
```

Comment:
Attributes such as password, email, confirmed_at, etc. are authentication details and should not be exposed to the user.